ui/store/actions.js

```diff
@@ -2199,6 +2199,21 @@ export function setSwapsTxGasLimit(gasLimit) {
   };
 }
 
+export function updateCustomSwapsEIP1559GasParams({
+  gasLimit,
+  maxFeePerGas,
+  maxPriorityFeePerGas,
+}) {
+  return async (dispatch) => {
+    await promisifiedBackground.setSwapsTxGasLimit(gasLimit);
+    await promisifiedBackground.setSwapsTxMaxFeePerGas(maxFeePerGas);
+    await promisifiedBackground.setSwapsTxMaxFeePriorityPerGas(
+      maxPriorityFeePerGas,
+    );
```

Comment:
Doesn't need to block merge but can this be a `Promise.all`, just so we aren't successively awaiting?

Reply:
That sounds promising!